flink-core/src/main/java/org/apache/flink/api/java/typeutils/runtime/TupleSerializerBase.java

```diff
@@ -56,7 +58,19 @@ public boolean isImmutableType() {
 
 	@Override
 	public int getLength() {
-		return -1;
+		if (length == -2) {
+			int sum = 0;
+			for (TypeSerializer<Object> serializer : fieldSerializers) {
+				if (serializer.getLength() != -1) {
```

Comment:
How about making this check `serializer.getLength() > 0`? That way, we catch cases of "misbehaved" subclasses.